presentation/src/main/java/com/pawel/presentation/ui/movieDetail/DetailMovieFragment.kt

```diff
@@ -63,26 +66,43 @@ class DetailMovieFragment : Fragment() {
     }
 
     private fun setOnMovieResponseObserver() {
+        /**
+         * EspressoIdlingResource...() utility for AndroidTest
+         */
         EspressoIdlingResource.increment()
-        viewModel.movie.observe(viewLifecycleOwner) {
-            it?.let {
-                movie = it
-                updateUITexts()
-                updateUIImageView(
-                    binding.detailMovieBackdrop.context,
-                    "${POSTER_URL}w500/${movie.backdrop_path}",
-                    binding.detailMovieBackdrop
-                )
-                updateUIImageView(
-                    binding.detailMoviePoster.context,
-                    "${POSTER_URL}w300/${movie.poster_path}",
-                    binding.detailMoviePoster
-                )
-
+        viewModel.networkResponse.observe(viewLifecycleOwner) { event ->
+            event.getContentIfNotHandled()?.let { result ->
+                when (result) {
+                    is SingleMovie -> {
+                        movie = result.movie
+                        updateUITexts()
+                        updateUIImageView(
+                            binding.detailMovieBackdrop.context,
+                            "${POSTER_URL}w500/${movie.backdrop_path}",
```

Comment:
You can move the creation of the full URL to the data layer
- move POSTER_URL to the same place as the base_url configuration
- do the converstion to the full URL when you're creating the response entity